grails-app/services/org/pih/warehouse/inventory/InventoryItemDataService.groovy

```diff
@@ -0,0 +1,15 @@
+package org.pih.warehouse.inventory
+
+import grails.gorm.services.Join
+import grails.gorm.services.Service
+
+@Service(InventoryItem)
+interface InventoryItemDataService {
+
+    InventoryItem save(InventoryItem inventoryItem);
+
+    @Join("product")
+    InventoryItem get(String id);
```

Comment:
That's because `lazyInitializationException` was thrown during getting a product from inventoryItem: `inventoryItem?.product?.id`

Reply:
Where you are getting this product?